### PR TITLE
Fix graphicbuilder transform

### PR DESCRIPTION
**Pull request by markschlosseratbentley**
Geometry created by a GraphicBuilder needs to have its specified placement transform applied properly.  This PR also modularizes some test functionality used by tests in frontend/src/test/render/webgl/

### CODE REVIEW

File: core/frontend/src/render/primitives/geometry/GeometryPrimitives.ts [hunk 1]
```diff
@@ -65,13 +65,23 @@ export abstract class Geometry {
     if (!this.displayParams.ignoreLighting) // ###TODO don't generate normals for 2d views.
       facetOptions.needNormals = true;
 
-    return this._getPolyfaces(facetOptions);
+    const polyfaceList = this._getPolyfaces(facetOptions);
+    if (undefined !== polyfaceList)
```
Comment (pmconne) [inline core/frontend/src/render/primitives/geometry/GeometryPrimitives.ts:69]: Avoid wasting time applying identity transform (typical case) here and in getStrokes.

**Review by pmconne — CHANGES_REQUESTED** [review]

**Review by pmconne — APPROVED** [review]

**Comment by markschlosseratbentley** [comment]
@mergifyio update

**Comment by pmconne** [comment]
@Mergifyio update

**Comment by markschlosseratbentley** [comment]
@Mergifyio backport release/2.19.x

**Pull request merged**